### PR TITLE
MAYA-122888 - Lights outliner icon is not getting picked up

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-122888 - Lights outliner icon is not getting picked up
* The Usd LuxLight schema was removed in previous USD version and replace with LuxBoundableLightBase.
* Created new UI test for the icon types to catch this kind of change in the future.

### CODE REVIEW

File: test/lib/ufe/CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +41,10 @@ if(CMAKE_UFE_V2_FEATURES_AVAILABLE)
         testUIInfoHandler.py
         testObservableScene.py
     )
+
+    list(APPEND INTERACTIVE_TEST_SCRIPT_FILES
+        testUIIcons.py
+    )
```
Comment (seando-adsk) [inline test/lib/ufe/CMakeLists.txt:47]: Add a new interactive test. I'll explain why its interactive below.

File: test/lib/ufe/testUIIcons.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        if swigIcon:
+            return wrapInstance(int(swigIcon), QIcon)
+        return None
+
+    def _getUIInfoIcon(self, primType):
+        primPath = ufe.PathString.path('%s,/%s' % (self.proxyShapePathStr, primType))
+        primItem = ufe.Hierarchy.createItem(primPath)
+        self.assertIsNotNone(primItem, msg='for prim path "%s"' % primPath)
+        ufeIcon = self.ufeUIInfo.treeViewIcon(primItem)
+        return ufeIcon.baseIcon if ufeIcon.baseIcon else ''
```
Comment (seando-adsk) [inline test/lib/ufe/testUIIcons.py:72]: Get the icon name (if one exists) from the Ufe UIInfoHandler.

File: test/lib/ufe/testUIIcons.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def _createIconFromFile(self, iconName):
+        swigIcon = mui.MQtUtil.createIcon(iconName)
+        if swigIcon:
+            return wrapInstance(int(swigIcon), QIcon)
+        return None
```
Comment (seando-adsk) [inline test/lib/ufe/testUIIcons.py:65]: Simple helper to try and create an icon from the input filename. This uses the MQtUtil which is why this tests needs to be interactive. We do this to mimic what happens in Maya (searching thru the XBMLANGPATH env var for a matching file).

File: test/lib/ufe/testUIIcons.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    def _getAncestorNodeIcon(self, primType):
+        primPath = ufe.PathString.path('%s,/%s' % (self.proxyShapePathStr, primType))
+        primItem = ufe.Hierarchy.createItem(primPath)
+        for ty in primItem.ancestorNodeTypes():
+            iconName = self.baseIconName + ty + '.png'
+            icon = self._createIconFromFile(iconName)
+            if icon:
+                return iconName
+        return ''
```
Comment (seando-adsk) [inline test/lib/ufe/testUIIcons.py:82]: Helper to mimic what the Outliner does when no icon from UIInfoHandler is found. Iterate thru the ancestor node types looking for a icon matching that name.

File: test/lib/ufe/testUIIcons.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            ])
+
+        for ty,iname in primTypes:
+            prim = self.stage.DefinePrim('/%s' % ty, ty)
+            ufeIcon = self._getUIInfoIcon(ty)
+            if ufeIcon:
+                self.assertEqual(ufeIcon, iname, msg='for prim type "%s"' % ty)
+            else:
+                iconFile = self._getAncestorNodeIcon(ty)
+                self.assertEqual(iconFile, iname, msg='for prim type "%s"' % ty)
```
Comment (seando-adsk) [inline test/lib/ufe/testUIIcons.py:156]: For each registered node type (including the new ones added in 21.11) create a prim of that type and then first ask the UIInfoHandler for the icon. If not found use the ancestor loop. In both cases assert that the icon found is equal to the one we expect.

**Review by ppt-adsk — APPROVED** [review]
Thanks for the test!

**Review by ppt-adsk — APPROVED** [review]

File: cmake/test.cmake [hunk 6]
```diff
@@ -228,8 +228,14 @@ finally:
          "${CMAKE_INSTALL_PREFIX}/lib/scripts")
     list(APPEND MAYAUSD_VARNAME_MAYA_SCRIPT_PATH
          "${CMAKE_INSTALL_PREFIX}/lib/scripts")
-    list(APPEND MAYAUSD_VARNAME_XBMLANGPATH
-         "${CMAKE_INSTALL_PREFIX}/lib/icons")
+    if (IS_LINUX)
+        # On Linux the paths in XBMLANGPATH need a /%B at the end.
+        list(APPEND MAYAUSD_VARNAME_XBMLANGPATH
+             "${CMAKE_INSTALL_PREFIX}/lib/icons/%B")
```
Comment (ppt-adsk) [inline cmake/test.cmake:234]: On the obscure side, I would say.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**